docs/faq.md

```diff
@@ -109,6 +109,46 @@ commands =
 `tox` will not do any install. Poetry installs all the dependencies and the current package an editable mode.
 Thus, tests are running against the local files and not the built and installed package.
 
+#### Usecase #4
+```ini
+[tox]
+isolated_build = true
+
+[testenv]
+allowlist_externals = poetry
+commands_pre =
+    poetry install --no-root --sync --with test
+commands =
+    pytest tests/
+
+[testenv:style]
+skip_install = true
+commands_pre =
+    poetry install --no-root --sync --with style
+commands =
+    flake8 src/
+    black --check src/
+
+[testenv:typing]
+commands_pre =
+    poetry install --no-root --sync --with typing
+commands =
+    mypy src/
+```
+
+`tox` will create an `sdist` package of the project and use `pip` to install it in a fresh environment.
+Thus, dependencies are resolved by `pip` in the first place. But afterwards we run Poetry, which will
+install the locked dependencies into the environment.
+
+In the `style` testenv, `tox` will only create the virtual environment, and
+again we will rely on Poetry to install requirements. The project itself will
+not be installed at all.
+
+`poetry` is used to sync package dependencies and named dependency groups into
+the `tox`-created virtual environments. This allows dependency groups to provide
+dependencies for different `tox` environments, providing `test` to the main
+testenv, `style` to the `style` env, and `typing` to the `typing` env.
+
```

Comment:
Maybe we should show the `pyproject.toml` here as well. You are consequently use the `--with` parameter for the groups, which means that these groups must be marked as optional.

I guess it would be nice to have some groups optional and some not. So we can show the difference between `--only` and `--with`.

Reply:
I like the idea of showing the `pyproject.toml`, and that would make me rethink what I said about calling this one a "Complex Example". If we add the pyproject file, it becomes much more complex to read than the others (but also more informative).

Can we split the difference by calling it "complex", omitting `pyproject.toml` for now, and I can try to add it in a follow-up PR?